### DISCUSSION TITLE
Uxarray Early Testing

### DISCUSSION THREAD

**Discussion by anissa111** [UXARRAY/uxarray#34] [General]
Hello uxarray community! 

As this effort gets off the ground, there will undoubtedly be some issues, bugs, things that we didn't consider. We know not all issues or use cases can be covered by unit tests and would therefore like to invite everyone, contributors and user community to kick our tires. See what works, what doesn't, and what could use some improvement before an official release.

If you want to help, just use the package! If you're looking for an example of how to get going, you can look at GeoCAT's collaborators internal testing repo, [here](https://github.com/NCAR/geocat-internal-testing/tree/main/uxarray).

We ask that, if appropriate, anything you find be brought to this discussion thread as a way of keeping everything localized and ready for easy discussion.

Thank you and happy bug hunt!!

**Comment by anissa111** [comment]
# Minimal Create From Vertex
As a way to start us off, I'll get us started with a quirk I found. The API allows for the creation of a `ux.Grid()` from vertices. Presumably this would require at least three vertices.

If you try to initialize with one point, an error is returned as expected. This
```python3
import uxarray as ux
import numpy as np

# initialize with one point
line = np.array([[0, 0]])
ux_line = ux.Grid(line)
```
will return `ValueError: different number of dimensions on data and dims: 0 vs 1`

However, if you try to initialize with two points, no error is given:
```python3
# initialize with two points
line = np.array([[0, 0], [1, 1]])
ux_line = ux.Grid(line)
```

Should we throw an error for this input and maybe provide a custom error for initialization with one point? Should we also check that the vertices provided are unique?

Let's talk about it!

**Comment by brianpm** [comment]
I installed uxarray and tried to load a mesh file that is used by CESM, but I guess uxarray can't read this format yet. It looks like it is an ESMF-generated mesh. Here's what I did:

```
import uxarray as uxr
mesh = uxr.Grid("/glade/p/cesmdata/cseg/inputdata/share/meshes/ne30pg3_ESMFmesh_cdf5_c20211018.nc")
```
Probably predictably, I get:
```
RuntimeError: /glade/p/cesmdata/cseg/inputdata/share/meshes/ne30pg3_ESMFmesh_cdf5_c20211018.nc is not a valid input type.
```

**Reply by erogluorhan** [reply]
Created the Issue #107 referring to this
